huaweicloud/services/as/resource_huaweicloud_as_planned_task.go

```diff
@@ -0,0 +1,367 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/autoscaling/v1/scheduledtasks"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+var (
+	PlannedTaskRecurrencePeriods = []string{"DAILY", "WEEKLY", "MONTHLY"}
+)
+
+func ResourcePlannedTask() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourcePlannedTaskCreate,
+		ReadContext:   resourcePlannedTaskRead,
+		UpdateContext: resourcePlannedTaskUpdate,
+		DeleteContext: resourcePlannedTaskDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: resourcePlannedTaskImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The region where the planned task resource are located.",
+			},
+			"scaling_group_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The ID of the scaling group where the planned task to create.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the planned task to create.",
+			},
+			"scheduled_policy": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"launch_time": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The execution time of planned task.",
+						},
+						"recurrence_type": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice(PlannedTaskRecurrencePeriods, false),
+							Description:  "The triggering type of planned task",
+						},
+						"recurrence_value": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The frequency at which planned task are triggered",
+						},
+						"start_time": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The effective start time of planned task.",
+						},
+						"end_time": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The effective end time of planned task",
+						},
+					},
+				},
+				Description: "The policy of planned task to create.",
+			},
+			"instance_number": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"max": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The maximum number of instances for the scaling group",
+						},
```

Comment:
```suggestion
						"max": {
							Type:        schema.TypeString,
							Optional:    true,
							Description: "The maximum number of instances for the scaling group",
							AtLeastOneOf: []string{"min", "desire"},
						},
```
maybe append an `AtleastOneOf`

Reply:
fixed.